src/webapp/pages/dashboard/usePerformanceOverview.ts

```diff
@@ -1,52 +1,62 @@
 import { useEffect, useState } from "react";
 import { useAppContext } from "../../contexts/app-context";
+import _ from "../../../domain/entities/generic/Collection";
 
-import { FilterType, TableColumn } from "../../components/table/statistic-table/StatisticTable";
-import { Id } from "../../../domain/entities/Ref";
+import { FiltersConfig, TableColumn } from "../../components/table/statistic-table/StatisticTable";
+import { ProgramIndicatorBaseAttrs } from "../../../data/repositories/AnalyticsD2Repository";
 
 type State = {
     columns: TableColumn[];
     dataPerformanceOverview: any[];
     columnRules: { [key: string]: number };
     editRiskAssessmentColumns: string[];
-    filters: FilterType[];
+    filters: FiltersConfig[];
+    order?: Order;
+    setOrder: (order: Order) => void;
     isLoading: boolean;
 };
 
-type PerformanceOverviewData = {
-    id: Id;
-    event: string;
-    location: string;
-    cases: string;
-    deaths: string;
-    duration: string;
-    manager: string;
-    detect7d: string;
-    notify1d: string;
-    era1: string;
-    era2: string;
-    era3: string;
-    era4: string;
-    era5: string;
-    era6: string;
-    era7: string;
-    eri: string;
-    respond7d: string;
-};
+export type Order = { name: string; direction: "asc" | "desc" };
+
 export function usePerformanceOverview(): State {
     const { compositionRoot } = useAppContext();
 
     const [dataPerformanceOverview, setDataPerformanceOverview] = useState<
-        PerformanceOverviewData[]
+        ProgramIndicatorBaseAttrs[]
     >([]);
     const [isLoading, setIsLoading] = useState(false);
+    const [order, setOrder] = useState<Order>();
+
+    useEffect(() => {
+        if (dataPerformanceOverview) {
+            setDataPerformanceOverview(newDataPerformanceOverview =>
+                _(newDataPerformanceOverview)
```

Comment:
whenever your set of a react state depends on its previous state, you need to use an updater function. https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state. Then add dataPerformanceOverviewdata as a dependency to the useEffect